### ISSUE TITLE
[NEWS] Wysyłam z Allegro - zmienimy typy pól / Ship with Allegro - we will change the field types

### ISSUE THREAD

**Issue by Lukasz-Zurek** [allegro/allegro-api#5970]
Zasoby /parcel-management pozwalają zarządzać przesyłkami w usłudze [Wysyłam z Allegro](https://developer.allegro.pl/tutorials/jak-zarzadzac-przesylkami-przez-wysylam-z-allegro-LRVjK7K21sY).

**Co zmienimy?**

3 października 2022 zmienimy typ pól. Obecną wartość z liczbą całkowitą (integer) zastąpimy typem tekstowym (string).

**Gdzie wprowadzamy zmiany?**

Typ wartości zmienimy dla:

- [GET /parcel-management/delivery-services](https://developer.allegro.pl/documentation/#operation/getAvailableDeliveryServices) - w zwracanym polu deliveryServices.id,
- [PUT /parcel-management/parcel-create-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/createNewParcel) - w polach:

  - serviceId,
  - Items.dimensions.height,
  - items.dimensions.width,
  - Items.dimensions.depth,

oraz w zwracanych polach:

-
  - input.serviceId,
  - input.Items.dimensions.height,
  - input.items.dimensions.width,
  - input.Items.dimensions.depth,
- [GET /parcel-management/parcels/{parcelId}](https://developer.allegro.pl/documentation/#operation/getParcelDetails) - w zwracanych polach:

  - serviceId,
  - Items.dimensions.height,
  - items.dimensions.width,
  - Items.dimensions.depth.

**Dlaczego wprowadzimy zmianę?**

Chcemy ujednolicić zwracane typy danych w naszym API, tak aby wszystkie wartości liczbowe były przekazywane w jednakowym formacie.

------

The /parcel-management resources allow you to manage shipments in the [Ship with Allegro](https://developer.allegro.pl/tutorials/how-to-manage-parcels-via-ship-with-allegro-ZM9YAyGKWTV) service.

**What will we change?**

On October 3, 2022, we will change the field types. We will replace the current integer value with a text type (string).

**Where are we making changes?**

We will change the value type for:

- [GET /parcel-management/delivery-services](https://developer.allegro.pl/documentation/#operation/getAvailableDeliveryServices) - in the returned deliveryServices.id field,
- [PUT /parcel-management/parcel-create-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/createNewParcel) - in the fields:

  - serviceId,
  - Items.dimensions.height,
  - items.dimensions.width,
  - Items.dimensions.depth,
  
and in the returned fields:

-
  - input.serviceId,
  - input.Items.dimensions.height,
  - input.items.dimensions.width,
  - input.Items.dimensions.depth,
- [GET /parcel-management/parcels/{parcelId}](https://developer.allegro.pl/documentation/#operation/getParcelDetails) - in the returned fields:

  - serviceId,
  - Items.dimensions.height,
  - items.dimensions.width,
  - Items.dimensions.depth.
  
**Why are we making a change?**

We want to standardize the return data types in our API so that all numeric values are provided in the same format.

**Comment by michal-dulik** [comment]
Czy można już przepinać wysyłane dane? lub jest możliwość przetestowania zmian przed 3 października?

**Comment by Lukasz-Zurek** [comment]
@michal-dulik Nie, ale możliwość wcześniejszego udostępnienia zmian na środowisku testowym skonsultuję z odpowiednim zespołem. Wrócę do Ciebie z informacją.

**Comment by Onixarts** [comment]
`GET https://api.allegro.pl/parcel-management/delivery-services` zwraca obecnie:

```json
{
    "deliveryServices": [
        {
            "id": "7741132",
            "service": "DPD",
            "name": "Allegro DPD",
            "owner": "ALLEGRO",
            "carrierId": "DPD",
```

A więc w chwili obecnej jest już zwracany id w typie string, podczas gdy w dokumentacji nadal widnieje, że jest to integer >=1. Czy zmiana typu oznacza zmianę stringa liczbowego na stringa guidowego?

**Comment by Lukasz-Zurek** [comment]
Zaktualizujemy to 3 października.

Dodam, że będzie to zmiana na stringa guidowego. Musi to być w zgodzie z [zapowiedzianą wczoraj zmianą](https://developer.allegro.pl/news/wysylam-z-allegro-zmienimy-identyfikatory-uslug-dostawy-BvMdq20KEIm).

**Comment by Lukasz-Zurek** [comment]
@michal-dulik Zmiana jest już od wczoraj na środowisku testowym.

**Comment by Onixarts** [comment]
> Zmiana jest już od wczoraj na środowisku testowym.

Obecnie odpowiedź endpointa na Allegro jak i sandboxie jest identyczna. W obu serwisach zwracana jest wartość liczbowa jako string - czym to się więc różni w sandboxie od produkcyjnego serwera i co zmieni się 3 października? Czy na sandboxie już powinny być GUIDy? Bo jeśli tak to u mnie nie ma, są nadal liczby.

**Comment by Lukasz-Zurek** [comment]
@Onixarts Jest to uwarunkowane tym, że Furgonetka.pl zwraca nam identyfikatory usług dostawy jako integer. Natomiast od 26.10 zarówno na produkcji, jak i na sandboxie - będą to już identyfikatory tekstowe/GUID.

**Comment by Onixarts** [comment]
Chyba się nie rozumiemy. Zapowiedzieliście zmiany na 3.10. Dzisiaj mamy 27.09 a zmiany te już są na Allegro. Nie chodzi o to, czy będzie to GUID czy liczba, chodzi o typ danych. Już dzisiaj jest to string a powinien być integer.

to jest string: `"123"`
to jest integer: `123`

w jsonie.

**Comment by Lukasz-Zurek** [comment]
@Onixarts Masz rację. Właśnie to wyjaśniam. Dziękuję Ci za zgłoszenie. Jednocześnie przepraszam za te utrudnienia.

**Comment by Onixarts** [comment]
Dziękuję, działa, świat uratowany, możemy iść do domu. :)

**Comment by Onixarts** [comment]
@Lukasz-Zurek jeszcze pytanko odnośnie `items.dimensions`. W tej chwili są tam obiekty `DimensionItem` składające się z 

```json
{
 "dimensions": {
    "width": { 
      "value": 23,
      "unit": "CENTIMETER"
    },
  "height": { 
      "value": 23,
      "unit": "CENTIMETER"
    },
  "depth": { 
      "value": 23,
      "unit": "CENTIMETER"
    }
}
```

Czy zamiast tych struktur pojawi się jeden string i co się stanie z jednostką `unit`? Czy zamiast nich pojawią się tylko same wartości stringowe?

```json
{ 
  "dimensions": {
   "height": "23",
   "width": "23",
   "depth": "23"
  }
}
```

**Comment by Lukasz-Zurek** [comment]
> @Lukasz-Zurek jeszcze pytanko odnośnie `items.dimensions`. W tej chwili są tam obiekty `DimensionItem` składające się z
> 
> ```json
> {
>  "dimensions": {
>     "width": { 
>       "value": 23,
>       "unit": "CENTIMETER"
>     },
>   "height": { 
>       "value": 23,
>       "unit": "CENTIMETER"
>     },
>   "depth": { 
>       "value": 23,
>       "unit": "CENTIMETER"
>     }
> }
> ```
> 
> Czy zamiast tych struktur pojawi się jeden string i co się stanie z jednostką `unit`? Czy zamiast nich pojawią się tylko same wartości stringowe?
> 
> ```json
> { 
>   "dimensions": {
>    "height": "23",
>    "width": "23",
>    "depth": "23"
>   }
> }
> ```
> 

Nie, struktura pozostanie bez zmian, tj. pola "unit" nadal będą dostępne.

> Czy zmienią się typy elementu `dimmensionItem.value`, czyli:
> 
> ```json
> {
>  "dimensions": {
>     "width": { 
>       "value": "23",
>       "unit": "CENTIMETER"
>     },
>   "height": { 
>       "value": "23",
>       "unit": "CENTIMETER"
>     },
>   "depth": { 
>       "value": "23",
>       "unit": "CENTIMETER"
>     }
> }
> ```

Tak, zmianie ulegną tylko pola "value".

**Comment by Onixarts** [comment]
@Lukasz-Zurek prosiłbym o informację, gdy zmiana zostanie opublikowana.

**Comment by Lukasz-Zurek** [comment]
@Onixarts Oczywiście dam znać jak tylko zmiana zostanie wdrożona.

**Comment by Lukasz-Zurek** [comment]
@Onixarts Zmianę już wdrożyliśmy.
